Mathlib/Data/List/Commutativity.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2025 Shimanonogov Igor. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shimanonogov Igor
+-/
+import Mathlib.Data.List.Defs
+import Mathlib.Data.List.RepeatSelf
+
+/-!
+# commutativity
+
+Results commencted with commutativity of List with respect to append.
```

Comment:
I would re-phrase this sentence as something like `This file contains results about commutativity of `List.append`. But if the main purpose of this file is to prove the L-S theorem; you should really mention that - perhaps "In this file we prove the L-S theorem, that (some description here). As a by-product, we prove various results about commutativity of appending to lists".

As a final comment, the word "commutativity" in the context of List.append is a bit odd: I don't find it obvious what this means. Describing the theorem statement could help here.